src/Testcontainers/Configurations/Modules/Databases/MySqlTestcontainerConfiguration.cs

```diff
@@ -39,15 +42,29 @@ public override string Database
     /// <inheritdoc />
     public override string Username
     {
-      get => this.Environments["MYSQL_USER"];
-      set => this.Environments["MYSQL_USER"] = value;
+      get {
+        return this._username;
+      }
+      set {
+        this._username = value;
+        if(value != RootUsername) {
```

Comment:
Do we really need to distinguish here? I think we need to make sure `MYSQL_ALLOW_EMPTY_PASSWORD` is set, then it does not matter. Without the if-else it would be easier to read.

Reply:
as we went ahead and set the password for both it should not hurt to also set "both" usernames.

Reply:
Nope, its not working. MySQL Image is complaining if the username is set to root. That's why I made this PR in the first place :D
```
2022-10-23 19:22:17+00:00 [ERROR] [Entrypoint]: MYSQL_USER="root", MYSQL_USER and MYSQL_PASSWORD are for configuring a regular user and cannot be used for the root user
    Remove MYSQL_USER="root" and use one of the following to control the root user password:
    - MYSQL_ROOT_PASSWORD
    - MYSQL_ALLOW_EMPTY_PASSWORD
    - MYSQL_RANDOM_ROOT_PASSWORD
```

Reply:
I refactored a little: [3aa809e](https://github.com/testcontainers/testcontainers-dotnet/pull/629/commits/3aa809ead984eaac32b08c927bb0bbb37d4741c8)

Reply:
Thanks for checking,